provider/identity_provider_resource.go

```diff
@@ -103,28 +110,138 @@ func (t *IdentityProviderResourceType) gitlabSchema() tfsdk.NestedAttributes {
 	return tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
 		"ca": {
 			Description: "Optional trusted certificate authority bundle.",
-			Type:     types.StringType,
-			Optional: true,
+			Type:        types.StringType,
+			Optional:    true,
 		},
 		"client_id": {
 			Description: "Client identifier of a registered Gitlab OAuth application.",
-			Type:     types.StringType,
-			Required: true,
+			Type:        types.StringType,
+			Required:    true,
 		},
 		"client_secret": {
 			Description: "Client secret issued by Gitlab.",
-			Type:      types.StringType,
-			Required:  true,
-			Sensitive: true,
+			Type:        types.StringType,
+			Required:    true,
+			Sensitive:   true,
 		},
 		"url": {
 			Description: "URL of the Gitlab instance.",
-			Type:     types.StringType,
-			Required: true,
+			Type:        types.StringType,
+			Required:    true,
+		},
+	})
+}
+
+func (t *IdentityProviderResourceType) githubSchema() tfsdk.NestedAttributes {
+	return tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+		"hostname": {
+			Description: "Optional domain to use with a hosted instance of GitHub Enterprise.",
+			Type:        types.StringType,
+			Optional:    true,
+		},
+		"organizations": {
+			Description: "Only users that are members of at least one of the listed organizations will be allowed to log in.",
+			Type: types.ListType{
+				ElemType: types.StringType,
+			},
+			Optional: true,
+		},
+		"teams": {
+			Description: "Only users that are members of at least one of the listed teams will be allowed to log in. The format is <org>/<team>.",
+			Type: types.ListType{
+				ElemType: types.StringType,
+			},
+			Optional: true,
+		},
+		"ca": {
+			Type:        types.StringType,
+			Description: "Path to PEM-encoded certificate file to use when making requests to the server.",
+			Optional:    true,
+		},
+		"client_id": {
+			Description: "Client identifier of a registered Github OAuth application.",
+			Type:        types.StringType,
+			Required:    true,
+		},
+		"client_secret": {
+			Description: "Client secret issued by Github.",
+			Type:        types.StringType,
+			Required:    true,
+			Sensitive:   true,
 		},
 	})
 }
 
+func (t *IdentityProviderResourceType) githubValidators() []tfsdk.AttributeValidator {
```

Comment:
I think that the method no need to be part of IdentityProviderResourceType struct
Later we will remove it to a different package.

Reply:
Will be moved to the new `idps` package